### PR TITLE
riscv: DW I2C driver for single master is ready to review

### PR DESCRIPTION
**Pull request by IanJiangICT**

### CODE REVIEW

File: arch/riscv/common/Kconfig [hunk 1]
```diff
@@ -401,4 +401,10 @@ config PLIC
 	bool "Platform level interrupt controller (PLIC) support"
 	depends ARCH_HAS_PLIC
 
+config ARCH_HAS_I2C
+	bool
```
Comment (zetalog) [inline arch/riscv/common/Kconfig]: Add this to kernel/i2c/Kconfig.

File: arch/riscv/common/Kconfig [hunk 2]
```diff
@@ -401,4 +401,10 @@ config PLIC
...
+
+config I2C
+	bool "Inter-Integrated Circuit (I2C) support"
```
Comment (zetalog) [inline arch/riscv/common/Kconfig]: This has already been defined in kernel/i2c/Kconfig.

File: drivers/i2c/dw_i2c_master.c [hunk 3]
```diff
@@ -1,3 +1,104 @@
...
+#endif
+	__raw_writel(0, base + IC_ENABLE);
+	__raw_writel((uint32_t)addr, base + IC_SAR);
+	__raw_writel(1, base + IC_ENABLE);
+}
+
+/* XXX */
+#ifndef CLK_CPU
+#define CLK_CPU 2000000000
+#endif
```
Comment (zetalog) [inline drivers/i2c/dw_i2c_master.c]: This seems unbelievably fast. Please confirm with your hardware spec.

File: drivers/i2c/Kconfig [hunk 4]
```diff
@@ -3,6 +3,9 @@ menu "Synopsys DesignWare I2C support"
 menuconfig DW_I2C
 	bool "Support DW I2C"
 
+config DW_I2C_CLK
+	int "Clock frequency (kHz)"
+	default 125000
```
Comment (zetalog) [inline drivers/i2c/Kconfig]: We normally prepare a definition, say DW_I2C_FREQ.
And define it in asm/mach/i2c.h before including driver/dw_i2c.h.

File: drivers/i2c/dw_i2c_master.c (hunk 3 above)
Comment (IanJiangICT) [reply drivers/i2c/dw_i2c_master.c]: Default is 125000 (kHz). Fixed.

File: arch/riscv/common/Kconfig (hunk 2 above)
Comment (IanJiangICT) [reply arch/riscv/common/Kconfig]: done

File: arch/riscv/common/Kconfig (hunk 1 above)
Comment (IanJiangICT) [reply arch/riscv/common/Kconfig]: done

**Pull request closed without merging**